src/mame/cirsa/neptunp2.cpp

```diff
@@ -244,8 +302,8 @@ void neptunp2_state::neptunp2_video(machine_config &config)
 	screen.set_refresh_hz(60);
 	screen.set_vblank_time(ATTOSECONDS_IN_USEC(0));
 	screen.set_screen_update(FUNC(neptunp2_state::screen_update));
-	screen.set_size(32*8, 32*8);
-	screen.set_visarea(0*8, 32*8-1, 2*8, 30*8-1);
+	screen.set_size(640, 480);
+	screen.set_visarea(0, 640-1, 0, 480-1);
```

Comment:
Is there a reason for this change?  The sizes are specified in terms of tiles, which makes sense for how the hardware works.  Lots of systems in MAME specify screen parameters in terms of tiles or CRTC parameter calculations (for example, src/mame/drivers/osborne1.cpp has both tile- and CRTC parameter-based screen parameter calculations).

Reply:
The manual says that it is "VGA resolution", that's why I changed it to 640x480.
This is the first dump for this hardware that includes the gfx SIMMs ROMs. Was the tile size calculated from the "neptunp2" game program ROM without having the gfx ROMs?

Reply:
I’m not sure, I was assuming the driver author was competent.  I guess it needs more of a detailed look.  But you’ve changed it to completely impossible values now – a CRT can’t work without horizontal and vertical blanking intervals, so the size needs to be larger than the visible area.  IIRC standard VGA is 800×525 total for 640×480 visible.

Reply:
If you’re going to change it to standard VGA values based on the manual, at least make a note that it’s an assumption based on that.

Reply:
`screen.set_raw(25.175_MHz_XTAL, 800, 0, 640, 524, 0, 480); // TODO: unconfirmed params, unknown clock`

Reply:
Made-up `set_raw` values with a crystal not present on the board is not a good idea.  Please stop recommending people do this.

Reply:
Made-up `screen.set_size` / `screen.set_visarea` is a better idea to you?

Reply:
To complete the comment: your request for a change here don't make any sense, and it's easy to spot out that isn't a valid screen geometry even without any comment. 
And without a valid clock changing the screen resolution to 800x525 won't change the fact that somebody else have to cover the full mile.

Reply:
If you set an estimate frame rate and vertical blanking interval, it’s clearer that it’s a guess.  You need to drop your obsession with changing everything to `set_raw` whether the actual pixel clock is known or not.  It’s not helping.

Reply:
Thanks a lot!!
I've left the values as they were, and added a comment.

Reply:
> If you set an estimate frame rate and vertical blanking interval, it’s clearer that it’s a guess. You need to drop your obsession with changing everything to `set_raw` whether the actual pixel clock is known or not. It’s not helping.

Look, the real problem here is `m_screen->set_vblank_time(ATTOSECONDS_IN_USEC(value));`, it's not the right abstraction for making screen stubs. 
It's ugly and nobody bother in making it right, causing all kinds of weird side effects later on and if the system has at least two screen irq events it basically sets a bomb under our seats. It's also pointless if you ever have to actually read vblank/hblank bits, particularly if `value` above is 0.

And anyway: my whole point was that I wouldn't have minded for the screen change as it was, not that if we actually have to use `set_raw` here.